TenancyInformationApi.Tests/V1/UseCase/ListTenanciesTests.cs

```diff
@@ -26,19 +28,84 @@ public void SetUp()
         [Test]
         public void GivenNoQueryParametersExecuteCallsTheGatewayToGetRecords()
         {
-            _mockGateway.Setup(x => x.ListTenancies()).Returns(new List<Tenancy>()).Verifiable();
-            _classUnderTest.Execute();
+            _mockGateway.Setup(x => x.ListTenancies(It.IsAny<int>(), It.IsAny<int>())).Returns(new List<Tenancy>()).Verifiable();
+            _classUnderTest.Execute(20, 0);
             _mockGateway.Verify();
         }
 
         [Test]
         public void ExecuteReturnsTenanciesFromTheGatewayMappedToResponses()
         {
-            var fixture = new Fixture();
-            var stubbedTenancies = fixture.CreateMany<Tenancy>().ToList();
-            _mockGateway.Setup(x => x.ListTenancies()).Returns(stubbedTenancies);
+            var stubbedTenancies = _fixture.CreateMany<Tenancy>().ToList();
+            _mockGateway.Setup(x => x.ListTenancies(It.IsAny<int>(), It.IsAny<int>())).Returns(stubbedTenancies);
 
-            _classUnderTest.Execute().Tenancies.Should().BeEquivalentTo(stubbedTenancies.ToResponse());
+            _classUnderTest.Execute(20, 0).Tenancies.Should().BeEquivalentTo(stubbedTenancies.ToResponse());
+        }
+
+        [Test]
+        public void ExecuteCallsTheGatewayWithLimitAndFormattedCursor()
+        {
+            _mockGateway.Setup(x => x.ListTenancies(23, 236712)).Returns(new List<Tenancy>()).Verifiable();
+            _classUnderTest.Execute(23, 236712);
+            _mockGateway.Verify();
+        }
+
+        [Test]
+        public void ExecuteIfNotCursorSuppliedPasses0ToTheGateway()
+        {
+            _mockGateway.Setup(x => x.ListTenancies(It.IsAny<int>(), 0)).Returns(new List<Tenancy>()).Verifiable();
+            _classUnderTest.Execute(20, 0);
+            _mockGateway.Verify();
+        }
+
+        [Test]
+        public void IfLimitLessThanTheMinimumWillUseTheMinimumLimit()
+        {
+            _mockGateway.Setup(x => x.ListTenancies(10, It.IsAny<int>()))
+                .Returns(new List<Tenancy>()).Verifiable();
+            _classUnderTest.Execute(0, 0);
```

Comment:
What would (0,0) return? Would it return nothing? Or use the default limit?

Reply:
It would return the minimum limit, which is 10.